fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoansApiResource.java

```diff
@@ -848,59 +862,121 @@ public String deleteLoanApplication(@PathParam("loanId") @ApiParam(value = "loan
-        } else if (is(commandParam, "disburseToSavings")) {
-            final CommandWrapper commandRequest = builder.disburseLoanToSavingsApplication(loanId).build();
+        }else if (is(commandParam, "approve")) {
+            final CommandWrapper commandRequest = builder.approveGLIMLoanApplication(glimId).build();
             result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
         }
-
-        if (is(commandParam, "undoapproval")) {
-            final CommandWrapper commandRequest = builder.undoLoanApplicationApproval(loanId).build();
-            result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
-        } else if (is(commandParam, "undodisbursal")) {
-            final CommandWrapper commandRequest = builder.undoLoanApplicationDisbursal(loanId).build();
+        else if (is(commandParam, "disburse")) {
+            final CommandWrapper commandRequest = builder.disburseGlimLoanApplication(glimId).build();
             result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
-        }else if (is(commandParam, "undolastdisbursal")) {
-            final CommandWrapper commandRequest = builder.undoLastDisbursalLoanApplication(loanId).build();
+        }  else if (is(commandParam, "glimrepayment")) {
+            final CommandWrapper commandRequest = builder.repaymentGlimLoanApplication(glimId).build();
             result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
         }
-
-        if (is(commandParam, "assignloanofficer")) {
-            final CommandWrapper commandRequest = builder.assignLoanOfficer(loanId).build();
+        else if (is(commandParam, "glimrepayment")) {
+            final CommandWrapper commandRequest = builder.repaymentGlimLoanApplication(glimId).build();
             result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
-        } else if (is(commandParam, "unassignloanofficer")) {
-            final CommandWrapper commandRequest = builder.unassignLoanOfficer(loanId).build();
+        }
+        else if (is(commandParam, "undodisbursal")) {
+            final CommandWrapper commandRequest = builder.undoGLIMLoanDisbursal(glimId).build();
             result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
-        } else if (is(commandParam, "recoverGuarantees")) {
-            final CommandWrapper commandRequest = new CommandWrapperBuilder().recoverFromGuarantor(loanId).build();
+        }
+        else if (is(commandParam, "undoapproval")) {
+            final CommandWrapper commandRequest = builder.undoGLIMLoanApproval(glimId).build();
             result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
         }
 
+
+        if (result == null) { throw new UnrecognizedQueryParamException("command", commandParam); }
```

Comment:
I think instead of throwing exception over here, we should let Client know about about incorrect param. 
By throwing exceptions, we sometimes reveal too much of inner mechanism to clients- just my thought.